src/manifest/mod.rs

```diff
@@ -36,6 +42,120 @@ struct CargoPackage {
     repository: Option<String>,
 }
 
+#[derive(Default, Deserialize)]
+struct CargoMetadata {
+    #[serde(default, rename = "wasm-pack")]
+    wasm_pack: CargoWasmPack,
+}
+
+#[derive(Default, Deserialize)]
+struct CargoWasmPack {
+    #[serde(default)]
+    profile: CargoWasmPackProfiles,
+}
+
+#[derive(Default, Deserialize)]
+struct CargoWasmPackProfiles {
+    #[serde(default)]
+    dev: CargoWasmPackProfile<Dev>,
+
+    #[serde(default)]
+    release: CargoWasmPackProfile<Release>,
+
+    #[serde(default)]
+    profiling: CargoWasmPackProfile<Profiling>,
```

Comment:
The trait-based approach here is a little difficult to follow, and it may run the risk of getting unwieldy as we add more and more configuration.

Could the fields of `CargoWasmPackProfile` perhaps all be `Option<T>` and the defaults are applied later when we have more context?

Reply:
I'm basically a little worried of getting to trapped up in trying to shoehorn all the default/deserialization logic into serde. I'm not much of a serde expert by any means, though, and Cargo's way of doing this (the one thing I have experience with) is pretty wordy and verbose as well. 

One option might be do to something like having all of these fields be `CargoWasmPackProfile` but they're all tagged with `#[serde(deserialize_with = "local_function_$default")]`. In `local_function_*` it would deserialize a `CargoWasmPackProfileWithOptions` which deserializes an struct with a bunch of options in it, returning a `CargoWasmPackProfile` which doesn't have any options and uses `unwrap_or` everywhere. 

(that may also be a good way to deal with cascading options that configure others if they're not otherwise configured too)